types/address/hash.go

```diff
@@ -0,0 +1,74 @@
+package address
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"reflect"
+	"sort"
+	"unsafe"
+
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// Len is the length of base addresses
+const Len = sha256.Size
+
+type Addressable interface {
+	Address() []byte
+}
+
+// Hash creates a new address from address type and key
+func Hash(typ string, key []byte) []byte {
+	hasher := sha256.New()
+	hasher.Write(unsafeStrToByteArray(typ))
+	th := hasher.Sum(nil)
```

Comment:
Let's cache this?

Reply:
It can be cached in a public key structure, not here. O you have some better idea?